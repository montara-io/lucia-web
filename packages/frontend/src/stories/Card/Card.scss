@import '../../styles/mixins';

.m-card {
  width: 100%;
  height: 100%;
  position: relative;
  box-shadow: 0 3px 6px #00000033;
  border-radius: 20px;

  .children-container {
    padding: 0.938rem 1.375rem;
  }

  .internal-button {
    @include internal-link;
  }

  .m-help-tooltip-icon {
    float: right;
    padding: 8px;
    margin: 8px;
    border-radius: 50%;
    height: 34px;
    width: 34px;
    transition: all 0.1s ease-in-out;

    img {
      height: 18px;
      width: 18px;
    }

    &.clickable:hover {
      cursor: pointer;
    }

    &:hover {
      transform: scale(1.1);
      box-shadow: 0 3px 6px $black033;
    }
  }
}
