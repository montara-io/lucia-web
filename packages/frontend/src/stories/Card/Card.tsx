import React, { ReactNode, useState } from 'react'
import HelpTooltip from '../HelpTooltip/HelpTooltip'
import InternalBtn, { InternalBtnProps } from '../InternalBtn/InternalBtn'
import './Card.scss'

export type CardProps = {
  internalLink?: InternalBtnProps
  helpLinkOnClick?: () => void
  helpLinkTooltip?: string | ReactNode
  helpLinkTooltipId?: string
  helpLinkTooltipPosition?: 'top' | 'left'
  children?: any
}

const Card: React.FunctionComponent<CardProps> = (props) => {
  const [showTooltip, setShowTooltip] = useState(false)

  return (
    <div className="m-card">
      {showTooltip && (
        <HelpTooltip helpTooltipPosition={props.helpLinkTooltipPosition}>
          {props.helpLinkTooltip}
        </HelpTooltip>
      )}
      {props.helpLinkTooltip && (
        <div
          className={`m-help-tooltip-icon${
            props.helpLinkOnClick ? ' clickable' : ''
          }`}
          data-testid="m-help-tooltip-icon"
          {...(props.helpLinkOnClick && { onClick: props.helpLinkOnClick })}
          onMouseEnter={() => setShowTooltip(true)}
          onMouseLeave={() => setShowTooltip(false)}
        >
          <img
            src="/assets/help-modal-q-mark-small.svg"
            className="icon-q-mark"
            {...(props.helpLinkTooltipId
              ? { id: props.helpLinkTooltipId }
              : {})}
            alt=""
          />
        </div>
      )}
      <div className="children-container">{props.children}</div>
      {props.internalLink && <InternalBtn {...props.internalLink} />}
    </div>
  )
}

export default Card
