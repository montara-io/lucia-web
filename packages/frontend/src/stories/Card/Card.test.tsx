import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import React from 'react'
import Card from './Card'

describe('Test Card Component', () => {
  it('Snapshot renders correctly', () => {
    const clickFn = jest.fn()
    expect(
      <Card helpLinkOnClick={clickFn} helpLinkTooltip="tooltip">
        <div>Test</div>
      </Card>,
    ).toMatchSnapshot()
  })

  it('Snapshot renders correctly with internal link', () => {
    expect(
      <Card
        internalLink={{
          text: 'more',
          callback: () => {},
        }}
      >
        <div>Test</div>
      </Card>,
    ).toMatchSnapshot()
  })
})
