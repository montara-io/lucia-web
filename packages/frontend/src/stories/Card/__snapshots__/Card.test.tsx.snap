// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card Component Snapshot renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="m-card"
      >
        <div
          class="m-help-tooltip-icon clickable"
          data-testid="m-help-tooltip-icon"
        >
          <img
            alt=""
            class="icon-q-mark"
            src="/assets/help-modal-q-mark-small.svg"
          />
        </div>
        <div
          class="children-container"
        >
          <div>
            Test
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="m-card"
    >
      <div
        class="m-help-tooltip-icon clickable"
        data-testid="m-help-tooltip-icon"
      >
        <img
          alt=""
          class="icon-q-mark"
          src="/assets/help-modal-q-mark-small.svg"
        />
      </div>
      <div
        class="children-container"
      >
        <div>
          Test
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Card Component Snapshot renders correctly with internal link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="m-card"
      >
        <div
          class="children-container"
        >
          <div>
            Test
          </div>
        </div>
        <button
          class="internal-button"
          data-testid="internal-button-undefined"
        >
          more
          <img
            alt="more"
            src="/assets/internal-link-arrow.svg"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="m-card"
    >
      <div
        class="children-container"
      >
        <div>
          Test
        </div>
      </div>
      <button
        class="internal-button"
        data-testid="internal-button-undefined"
      >
        more
        <img
          alt="more"
          src="/assets/internal-link-arrow.svg"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
